dali/operators/nvcvop/nvcvop.h

```diff
@@ -0,0 +1,183 @@
+#define DALI_OPERATORS_NVCVOP_NVCVOP_H_
+
+
+#include <nvcv/DataType.h>
+#include <nvcv/BorderType.h>
+#include <string>
+#include <vector>
+#include <nvcv/Tensor.hpp>
+#include <nvcv/ImageBatch.hpp>
+
+#include "dali/kernels/dynamic_scratchpad.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/operator/sequence_operator.h"
+
+namespace dali::nvcvop {
+
+/**
+ * @brief Get the nvcv border mode from name
+ *
+ * @param border_mode border mode name
+ */
+NVCVBorderType GetBorderMode(const std::string &border_mode);
+
+/**
+ * @brief Get nvcv data kind of a given data type
+ */
+nvcv::DataKind GetDataKind(DALIDataType dtype);
+
+/**
+ * @brief Get image format for an image with a given data type and number of channels.
+ *
+ * The memory layout of the image is assumed to be contiguous, with interleaved, equally-sized channels.
+ */
+nvcv::ImageFormat GetImageFormat(DALIDataType dtype, int num_channels);
+
+/**
+ * @brief Wrap a sample view into an nvcv Image
+ *
+ * @param sample sample view
+ * @param format image format. It needs to match the shape and data type of the sample view
+ */
+nvcv::Image AsImage(SampleView<GPUBackend> sample, const nvcv::ImageFormat &format);
+
+/**
+ * @brief Wrap a const sample view into an nvcv Image
+ *
+ * @param sample sample view
+ * @param format image format. It needs to match the shape and data type of the sample view
+ */
+nvcv::Image AsImage(ConstSampleView<GPUBackend> sample, const nvcv::ImageFormat &format);
+
+/**
+ * @brief Allocates an image batch using a dynamic scratchpad.
+ * Allocated images have the shape and data type matching the samples in the given TensorList.
+ *
+ * It can be used to allocate a workspace batch that matches a given input batch.
+ */
+void AllocateImagesLike(const TensorList<GPUBackend> &t_list,
+                        kernels::DynamicScratchpad &scratchpad, nvcv::ImageBatchVarShape &output);
+
+/**
+ * @brief Wrap samples from the given TensorList into nvcv Images and push them to an output Image batch
+ */
+void PushImagesToBatch(const TensorList<GPUBackend> &t_list, nvcv::ImageBatchVarShape &batch);
+
+/**
+ * @brief A base class for the CVCUDA operators.
+ * It adds convenience methods to access inputs, outputs and arguments as nvcv types.
+ */
+template <typename BaseOp>
+class NVCVOperator: public BaseOp {
+ public:
+  using BaseOp::spec_;
+  using BaseOp::BaseOp;
+
+  /**
+   * @brief Get an operator argument as an nvcv Tensor.
+   * This method uses the dynamic scratchpad to allocate the (pinned and device) buffers
+   *  used by the output tensor.
+   *
+   * @tparam T data type of the DALI operator argument
+   * @tparam DTYPE expected nvcv Tensor data type
+   * @param arg_shape shape of the DALI operator argument
+   */
+  template <typename T, NVCVDataType DTYPE>
+  nvcv::Tensor AcquireTensorArgument(Workspace &ws, kernels::DynamicScratchpad &scratchpad,
```

Comment:
I guess we could just use an interface here - it doesn't matter if the scratchpad is dynamic or not. What matters (and should be documented!) is that the scratchpad returns its memory in an order that's compatible with `ws.stream()`.
```suggestion
  nvcv::Tensor AcquireTensorArgument(Workspace &ws, kernels::Scratchpad &scratchpad,
```

Reply:
Why was it marked as resolved? The DynamicScratchpad is still there - is it required or we could just use the Scratchpad interface?